packages/runtimes/hmr/src/loaders/hmr-runtime.js

```diff
@@ -154,10 +157,24 @@ if ((!parent || !parent.isParcelRequire) && typeof WebSocket !== 'undefined') {
 
         await hmrApplyUpdates(assets);
 
-        for (var i = 0; i < assetsToAccept.length; i++) {
-          var id = assetsToAccept[i][1];
+        // Dispose all old assets.
+        for (let i = 0; i < assetsToDispose.length; i++) {
+          let id = assetsToDispose[i][1];
+
+          if (!acceptedAssets[id]) {
+            hmrDispose(assetsToDispose[i][0], id);
+            acceptedAssets[id] = true;
+          }
+        }
+
+        // Run accept callbacks. This will also re-execute other disposed assets in topological order.
+        acceptedAssets = {};
```

Comment:
I think `acceptedAssets` can be a local variable now? Kind of confusing that it's both `disposedAssets` and `acceptedAssets` at the same time (and as a global variable for no apparent reason)